app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala

```diff
@@ -0,0 +1,165 @@
+                                                       val confirmForeignPropertyReportingMethod: ConfirmForeignPropertyReportingMethod,
+                                                       val confirmSEReportingMethod: ConfirmSEReportingMethod,
+                                                       val confirmUKPropertyReportingMethod: ConfirmUKPropertyReportingMethod,
+                                                       val incomeSourceDetailsService: IncomeSourceDetailsService,
+                                                       val retrieveBtaNavBar: NavBarPredicate)
+                                                      (implicit val ec: ExecutionContext,
+                                                      implicit override val mcc: MessagesControllerComponents,
+                                                      val appConfig: FrontendAppConfig) extends ClientConfirmedController
+  with FeatureSwitching {
+
+  def show(id: String, taxYear: String, changeTo: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(
+        id = id,
+        isAgent = true,
+        taxYear = taxYear,
+        changeTo = changeTo,
+        itvcErrorHandler = itvcErrorHandler
+      )
+  }
+
+  def showAgent(id: String, taxYear: String, changeTo: String): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+          implicit mtdItUser =>
+            handleRequest(
+              id = id,
+              isAgent = true,
+              taxYear = taxYear,
+              changeTo = changeTo,
+              itvcErrorHandler = itvcErrorHandler
+            )
+        }
+  }
+
+  private def handleRequest(id: String,
+                    isAgent: Boolean,
+                    taxYear: String,
+                    changeTo: String,
+                    itvcErrorHandler: ShowInternalServerError)
+                   (implicit user: MtdItUser[_]): Future[Result] = {
+
+    lazy val isSoleTraderBusiness = user.incomeSources.businesses.exists(business =>
+      business.incomeSourceId.contains(id) && !business.isCeased)
+
+    lazy val isForeignProperty = user.incomeSources.properties.exists(property =>
+      property.incomeSourceId.contains(id) && property.isForeignProperty && !property.isCeased)
+
+    lazy val isUKProperty = user.incomeSources.properties.exists(property =>
+      property.incomeSourceId.contains(id) && property.isUkProperty && !property.isCeased)
+
+    Future(
+      (isEnabled(IncomeSources), getTaxYearStartYearEndYear(taxYear), getReportingMethodKey(changeTo)) match {
+        case (false, _, _) =>
+          Ok(
+            customNotFoundErrorView()
+          )
+        case (_, _, None) =>
```

Comment:
we might don't really need this case, as this all will be covered by the last case defined in th